includes/DataObjects/Request.php

```diff
@@ -256,6 +259,11 @@ public function hasComments()
 		if ($this->hasCommentsResolved) {
 			return $this->hasComments;
 		}
+        
+		if (isset(self::$commentsResults[$this->id])) {
+		    $this->hasCommentsResolved = true;
+		    return $this->hasComments = self::$commentsResults[$this->id];
+		}
```

Comment:
I may have interpreted the code incorrectly, but each request has it's own class correct?  The static variable preserves the information across all the classes, so if another duplicate class is opened it forgoes the querying and returns the information already cached.  It may be redundant perhaps, but it shouldn't impact performance and operation.